ThingIFSDK/ThingIFSDK/Aggregation.swift

```diff
@@ -58,12 +58,20 @@ open class Aggregation: NSCoding {
     let function: FunctionType
 
 
-    private init(
+    private init?(
       _ function: FunctionType,
       field: String,
       fieldType: FieldType)
     {
-        fatalError("TODO: implement me.")
+        if function == .max || function == .sum || function == .min ||
+             function == .mean {
+            if fieldType != .integer && fieldType != .decimal {
+                return nil
```

Comment:
I don't think it is good only return nil when fieldType is wrong. It is better to check the fieldType in `ThingIFAPI`, where can return error.